pennylane/_qutrit_device.py

```diff
@@ -80,9 +87,64 @@ def capabilities(cls):
         capabilities.update(model="qutrit")
         return capabilities
 
+    # Added here for tests to pass (See tests.test_qutrit_device.ExtractStatistics.test_results_no_state())
+    # The test deleted the state property of QutritDevice to test if the appropriate error is raised by
+    # `QutritDevice.statistics()` when requesting the state as a return type, but was failing as the test wouldn't
+    # actually delete the state attribute, as it was only explicitly defined in `QubitDevice`.
+    @property
+    def state(self):
+        """Returns the state vector of the circuit prior to measurement.
+
+        .. note::
+
+            Only state vector simulators support this property. Please see the
+            plugin documentation for more details.
+        """
+        raise NotImplementedError
+
     def statistics(self, observables, shot_range=None, bin_size=None):
         # Overloading QubitDevice.statistics() as VnEntropy and MutualInfo not yet supported for QutritDevice
-        raise NotImplementedError
+        results = []
+
+        for obs in observables:
+            # Pass instances directly
+            if obs.return_type is Expectation:
+                results.append(self.expval(obs, shot_range=shot_range, bin_size=bin_size))
+
+            elif obs.return_type is Variance:
+                results.append(self.var(obs, shot_range=shot_range, bin_size=bin_size))
+
+            elif obs.return_type is Sample:
+                results.append(self.sample(obs, shot_range=shot_range, bin_size=bin_size))
+
+            elif obs.return_type is Probability:
+                results.append(
+                    self.probability(wires=obs.wires, shot_range=shot_range, bin_size=bin_size)
+                )
+
+            elif obs.return_type is State:
+                if len(observables) > 1:
+                    raise qml.QuantumFunctionError(
+                        "The state or density matrix cannot be returned in combination"
+                        " with other return types"
+                    )
+                if self.wires.labels != tuple(range(self.num_wires)):
+                    raise qml.QuantumFunctionError(
+                        "Returning the state is not supported when using custom wire labels"
+                    )
+                # Check if the state is accessible and decide to return the state or the density
+                # matrix.
+                results.append(self.access_state(wires=obs.wires))
+
+            # This block covers the case when VnEntropy or MutualInfo are the return type for now
+            # Once VnEntropy and MutualInfo support is added for QutritDevice, condition blocks for
+            # them will be added separately.
+            elif obs.return_type is not None:
+                raise qml.QuantumFunctionError(
+                    f"Unsupported return type specified for observable {obs.name}"
+                )
```

Comment:
@antalszava @theodotk If I remove `QutritDevice.statistics`, I need to update `QutritDevice.sample` to support the counts functionality. However, doing so causes `TestSample.test_samples_with_bins` in `tests/test_qutrit_device.py` to fail due to an error thrown when reshaping the samples when `bin_size` is not `None` (see `pennylane/_qutrit_device.py:391`).

I'm not entirely certain I understand why the samples are reshaped to `(3, bin_size, -1)` in `QubitDevice.sample`, more specifically, why the first axis has shape 3, and I'd appreciate if I could get some insight into why so that I can update the code accordingly for `QutritDevice`.

Thanks in advance!

Reply:
I just wrote some test code for `default.qubit` using the new `Counts` measurement type, and it seems that it doesn't work when a shot vector is specified for the device and the number of wires being sampled is anything other than 3. I was about to open an issue but I just found [this issue](https://github.com/PennyLaneAI/pennylane/issues/2828), which points out the same problem. I have an idea for how to address this, and I have added a comment to the issue with my solution. I will also include my solution in `QutritDevice.sample()`, and add tests to `tests/test_qutrit_device.py` accordingly.

Reply:
Hi @mudit2812, thank you, this is really helpful! There are a couple of shortcomings of using shot vectors that we know about (and working to resolve). The issue with the number of wires having to be 3 is a separate bug. I'll be creating a fix for it for the `QubitDevice` case. :+1: